In [1]:
# 모듈 추가 
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv('교육부_학생건강검사.csv')
print(df.shape)   # 형태 확인 (행, 열)
print(df.head())

(9686, 25)
    고유식별번호(ID)       최종가중치 학교고유번호(ID)      도시규모 도시규모별분석용   학년도 광역시도       시도별  \
0  Aa011남10101  169.550665       Aa01  대도시/중소도시   특별/광역시  2015   서울  서울특별시교육청   
1  Aa011남10102  169.550665       Aa01  대도시/중소도시   특별/광역시  2015   서울  서울특별시교육청   
2  Aa011남10103  169.550665       Aa01  대도시/중소도시   특별/광역시  2015   서울  서울특별시교육청   
3  Aa011남10104  169.550665       Aa01  대도시/중소도시   특별/광역시  2015   서울  서울특별시교육청   
4  Aa011남10105  169.550665       Aa01  대도시/중소도시   특별/광역시  2015   서울  서울특별시교육청   

   학교급별       학교명  ...      키   몸무게  혈당식전(mgdl) 총콜레스테롤(mgdl) 간기능(AST)단위(UL)  \
0     1  서울대도초등학교  ...  125.8  27.3         NaN          NaN            NaN   
1     1  서울대도초등학교  ...  124.3  25.4         NaN          NaN            NaN   
2     1  서울대도초등학교  ...  119.2  23.5         NaN          NaN            NaN   
3     1  서울대도초등학교  ...  115.0  20.0         NaN          NaN            NaN   
4     1  서울대도초등학교  ...  120.0  33.5         NaN          NaN            NaN   

   간기능(ALT)단위(UL)  혈색소(gdl)

* 일단 불필요한 데이터 제거 전에 데이터 정보 확인

In [3]:
df.info()   # 데이터 정보 확인 - 결측치 확인 및 데이터 타입 확인

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9686 entries, 0 to 9685
Data columns (total 25 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   고유식별번호(ID)      9686 non-null   object 
 1   최종가중치           9686 non-null   float64
 2   학교고유번호(ID)      9686 non-null   object 
 3   도시규모            9686 non-null   object 
 4   도시규모별분석용        9686 non-null   object 
 5   학년도             9686 non-null   int64  
 6   광역시도            9686 non-null   object 
 7   시도별             9686 non-null   object 
 8   학교급별            9686 non-null   int64  
 9   학교명             9686 non-null   object 
 10  공학여부            9686 non-null   object 
 11  학년              9686 non-null   int64  
 12  반               9686 non-null   int64  
 13  성별              9686 non-null   object 
 14  건강검진일           9686 non-null   object 
 15  키               9686 non-null   float64
 16  몸무게             9682 non-null   float64
 17  혈당식전(mgdl)      338 non-null    f

In [4]:
df.describe()   # 기술통계 확인

최종가중치     학년도         학교급별           학년            반  \
count  9686.000000  9686.0  9686.000000  9686.000000  9686.000000   
mean    104.894487  2015.0     1.995767     2.567933     2.659096   
std      39.376988     0.0     0.856499     1.414243     2.601067   
min      23.325794  2015.0     1.000000     1.000000     1.000000   
25%      76.966829  2015.0     1.000000     1.000000     1.000000   
50%      99.963287  2015.0     2.000000     2.000000     1.000000   
75%     133.231158  2015.0     3.000000     3.000000     3.000000   
max     248.300844  2015.0     3.000000     6.000000    15.000000   

                 키          몸무게  혈당식전(mgdl)  총콜레스테롤(mgdl)  간기능(AST)단위(UL)  \
count  9686.000000  9682.000000   338.00000    339.000000      336.000000   
mean    154.564444    50.207591    92.14497    173.516519       27.824405   
std      16.968590    17.075953    10.49571     32.567571       19.327395   
min     107.700000    15.900000    65.00000    102.000000       11.000000   
25%     142.000000    37.200000    85.00000    148.500000       19.000000   
50%     158.700000    50.700000    91.00000    170.000000       24.000000   
75%     167.200000    60.600000    99.00000    191.000000       30.000000   
max     194.200000   130.700000   132.00000    300.000000      186.000000   

       간기능(ALT)단위(UL)    혈색소(gdl)          수축기          이완기  
count      337.000000  578.000000  3025.000000  3025.000000  
mean        30.311573   12.976990   104.830083    62.987438  
std         32.191960    1.092454    12.661348     7.948402  
min          8.000000    9.000000    65.000000    23.000000  
25%         14.000000   12.400000    98.000000    60.000000  
50%         20.000000   13.200000   105.000000    62.000000  
75%         32.000000   13.675000   112.000000    69.000000  
max        272.000000   15.700000   180.000000    98.000000

In [5]:
# 필요한 컬럼 추출을 하기 전에 결측치를 확인
df.isnull().sum()   # 결측치 확인

고유식별번호(ID)           0
최종가중치                0
학교고유번호(ID)           0
도시규모                 0
도시규모별분석용             0
학년도                  0
광역시도                 0
시도별                  0
학교급별                 0
학교명                  0
공학여부                 0
학년                   0
반                    0
성별                   0
건강검진일                0
키                    0
몸무게                  4
혈당식전(mgdl)        9348
총콜레스테롤(mgdl)      9347
간기능(AST)단위(UL)    9350
간기능(ALT)단위(UL)    9349
혈색소(gdl)          9108
간염검사              8855
수축기               6661
이완기               6661
dtype: int64

위에 생성된 데이터를 확인 했을 때 키 몸무게 외에 다른 측정 값들은 전체9686 중에 빈 데이터가 많아서 제거하는 방법을 선택해야 할 것으로 판단됨
- 사용하지 않을 컬럼 제거 
- 빈 데이터가 4개 있는 몸무게 정보를 채우거나 제거하기  

In [6]:
# 실제 사용할 데이터만 추출 
data = df.loc[:, ['학교고유번호(ID)','광역시도',	'학교명',	'학년',	'성별',	'키',	'몸무게']]
data.head()

학교고유번호(ID) 광역시도       학교명  학년 성별      키   몸무게
0       Aa01   서울  서울대도초등학교   1  남  125.8  27.3
1       Aa01   서울  서울대도초등학교   1  남  124.3  25.4
2       Aa01   서울  서울대도초등학교   1  남  119.2  23.5
3       Aa01   서울  서울대도초등학교   1  남  115.0  20.0
4       Aa01   서울  서울대도초등학교   1  남  120.0  33.5

In [7]:
# 컬럼 이름 변경 
data.rename(columns={'학교고유번호(ID)':'학교고유번호'}, inplace=True)
data.head()

학교고유번호 광역시도       학교명  학년 성별      키   몸무게
0   Aa01   서울  서울대도초등학교   1  남  125.8  27.3
1   Aa01   서울  서울대도초등학교   1  남  124.3  25.4
2   Aa01   서울  서울대도초등학교   1  남  119.2  23.5
3   Aa01   서울  서울대도초등학교   1  남  115.0  20.0
4   Aa01   서울  서울대도초등학교   1  남  120.0  33.5

In [8]:
#BMI 항목 추가
data['BMI'] = data['몸무게'] / (data['키'] / 100) ** 2
data.head()

학교고유번호 광역시도       학교명  학년 성별      키   몸무게        BMI
0   Aa01   서울  서울대도초등학교   1  남  125.8  27.3  17.250487
1   Aa01   서울  서울대도초등학교   1  남  124.3  25.4  16.439608
2   Aa01   서울  서울대도초등학교   1  남  119.2  23.5  16.539232
3   Aa01   서울  서울대도초등학교   1  남  115.0  20.0  15.122873
4   Aa01   서울  서울대도초등학교   1  남  120.0  33.5  23.263889

In [9]:
# 유일한 데이터 추출 
data['학교고유번호'].unique()  

array(['Aa01', 'Aa02', 'Aa03', 'Aa04', 'Aa06', 'Aa07', 'Aa08', 'Aa09',
       'Aa10', 'Aa11', 'Aa12', 'Aa13', 'Aa14', 'Aa15', 'Aa16', 'Aa17',
       'Aa18', 'Aa19', 'Aa20', 'Aa21', 'Aa22', 'Aa23', 'Aa24', 'Aa25',
       'Ab01', 'Ab02', 'Ab03', 'Ab04', 'Ab05', 'Ab06', 'Ab07', 'Ab08',
       'Ab09', 'Ab10', 'Ab11', 'Ab12', 'Ab13', 'Ab14', 'Ab15', 'Ab16',
       'Ab17', 'Ab19', 'Ab20', 'Ab21', 'Ab22', 'Ab23', 'Ab24', 'Ab25',
       'Ab26', 'Ab27', 'Ab28', 'Ab29', 'Ab30', 'Ac01', 'Ac02', 'Ac03',
       'Ac04', 'Ac05', 'Ac07', 'Ac08', 'Ac09', 'Ac10', 'Ac11', 'Ac12',
       'Ac13', 'Ac14', 'Ac15', 'Ac16', 'Ac17', 'Ac18', 'Ac19', 'Ac20',
       'Ac21', 'Ac22', 'Ac23', 'Ac24', 'Ac25', 'Ac26', 'Ac27', 'Ac28',
       'Ac29', 'Ac30', 'Ac31', 'Ac32', 'Ac33', 'Ac34'], dtype=object)

위에 출력 내용에 따른 데이터를 확인하면 Aa로 시작하면 초등학교, Ab로 시작하면 중학교, Ac로 시작하면 고등학교인 것을 학인할 수 있다. 

In [10]:
# 유일한 데이터 추출 
data['광역시도'].unique()


array(['서울'], dtype=object)

In [11]:
# 유일한 데이터 추출 
data['학교명'].unique()

array(['서울대도초등학교', '서울은빛초등학교', '서울구일초등학교', '서울길원초등학교', '서울원촌초등학교',
       '서울면동초등학교', '서울성내초등학교', '서울구산초등학교', '서울강명초등학교', '서울삼전초등학교',
       '서울난향초등학교', '서울천호초등학교', '서울독산초등학교', '서울쌍문초등학교', '서울영원초등학교',
       '경기초등학교', '서울가원초등학교', '서울중목초등학교', '서울신구초등학교', '서울월천초등학교',
       '서울삼광초등학교', '서울백산초등학교', '서울풍성초등학교', '서울창천초등학교', '정신여자중학교',
       '숙명여자중학교', '서울여자중학교', '배명중학교', '휘문중학교', '동북중학교', '경신중학교', '고척중학교',
       '양진중학교', '금옥중학교', '진관중학교', '한천중학교', '창동중학교', '창일중학교', '상봉중학교',
       '용강중학교', '창북중학교', '문정중학교', '구룡중학교', '불광중학교', '도곡중학교', '대림중학교',
       '강일중학교', '방학중학교', '고덕중학교', '한산중학교', '신연중학교', '태릉중학교', '관악중학교',
       '숙명여자고등학교', '대진여자고등학교', '동명여자고등학교', '동국대학교사범대학부속여자고등학교',
       '홍익대학교사범대학부속여자고등학교', '성심여자고등학교', '신정여자상업고등학교', '예일디자인고등학교',
       '서라벌고등학교', '동북고등학교', '대일고등학교', '문일고등학교', '용산고등학교', '면목고등학교',
       '오산고등학교', '영등포고등학교', '명지고등학교', '창동고등학교', '덕수고등학교', '불암고등학교',
       '선화예술고등학교', '서울세종고등학교', '청담고등학교', '한영외국어고등학교', '강동고등학교', '세현고등학교',
       '신서고등학교', '인덕공업고등학교', '서울문화고등학교', '서

In [12]:
# 유일한 데이터 추출
data['학년'].unique()

array([1, 2, 3, 4, 5, 6], dtype=int64)

지금 데이터 내용을 봐서는 학교별, 초중고별, 남여별로 분석이 가능 + 학년별 

In [13]:
# 특정 문자열을 포함하는 경우 추출 (학교고유번호가 Aa로 시작하는 경우 - 초등학교만 추출)
elementary = data[data['학교고유번호'].str.startswith('Aa')]
print(elementary.shape)   # 형태 확인 (행, 열)
print(elementary.head())

(3573, 8)
  학교고유번호 광역시도       학교명  학년 성별      키   몸무게        BMI
0   Aa01   서울  서울대도초등학교   1  남  125.8  27.3  17.250487
1   Aa01   서울  서울대도초등학교   1  남  124.3  25.4  16.439608
2   Aa01   서울  서울대도초등학교   1  남  119.2  23.5  16.539232
3   Aa01   서울  서울대도초등학교   1  남  115.0  20.0  15.122873
4   Aa01   서울  서울대도초등학교   1  남  120.0  33.5  23.263889


In [14]:
# 학교고유번호가 Ab로 시작하는 경우 - 중학교만 추출
middle = data[data['학교고유번호'].str.startswith('Ab')]
print(middle.shape)   # 형태 확인 (행, 열)
print(middle.head())

(2581, 8)
     학교고유번호 광역시도      학교명  학년 성별      키   몸무게        BMI
3573   Ab01   서울  정신여자중학교   1  여  153.7  50.0  21.165195
3574   Ab01   서울  정신여자중학교   1  여  156.8  46.5  18.913018
3575   Ab01   서울  정신여자중학교   1  여  160.7  47.9  18.548285
3576   Ab01   서울  정신여자중학교   1  여  161.8  51.0  19.481085
3577   Ab01   서울  정신여자중학교   1  여  157.3  45.0  18.186749


In [15]:
# 학교고유번호가 Ac로 시작하는 경우 - 중학교만 추출
high = data[data['학교고유번호'].str.startswith('Ac')]
print(high.shape)   # 형태 확인 (행, 열)
print(high.head())

(3532, 8)
     학교고유번호 광역시도       학교명  학년 성별      키   몸무게        BMI
6154   Ac01   서울  숙명여자고등학교   1  여  162.3  81.7  31.015945
6155   Ac01   서울  숙명여자고등학교   1  여  161.8  59.6  22.766131
6156   Ac01   서울  숙명여자고등학교   1  여  162.3  51.0  19.361239
6157   Ac01   서울  숙명여자고등학교   1  여  165.2  44.3  16.232434
6158   Ac01   서울  숙명여자고등학교   1  여  166.5  51.8  18.685352


고등학교 데이터에 대해 분석

In [16]:
high_count = high['학교명'].value_counts()
print(high_count)

명지고등학교               219
서울세종고등학교             218
불암고등학교               204
청담고등학교               201
강동고등학교               135
숙명여자고등학교             121
창동고등학교               120
서라벌고등학교              116
선화예술고등학교             115
대진여자고등학교             113
홍익대학교사범대학부속여자고등학교    109
동명여자고등학교             106
동국대학교사범대학부속여자고등학교    103
용산고등학교               102
오산고등학교               100
성심여자고등학교              99
동북고등학교                99
문일고등학교                96
신서고등학교                95
대일고등학교                94
세현고등학교                87
영등포고등학교               85
한영외국어고등학교             81
예일디자인고등학교             79
면목고등학교                79
서울문화고등학교              75
은평메디텍고등학교             75
서울금융고등학교              74
신정여자상업고등학교            74
인덕공업고등학교              74
덕수고등학교                73
고명경영고등학교              64
세종과학고등학교              47
Name: 학교명, dtype: int64


In [17]:
high_count = high['학년'].value_counts()
print(high_count)

2    1211
3    1166
1    1155
Name: 학년, dtype: int64


In [18]:
high_count = high['성별'].value_counts()
print(high_count)

여    1777
남    1755
Name: 성별, dtype: int64
